app/apphandlers/embeddedAppHandler.go

```diff
@@ -359,3 +372,19 @@ func (a *embeddedApp) StartRudderCore(ctx context.Context, options *app.Options)
 
 	return g.Wait()
 }
+
+func setupPipelineEnrichers(conf *config.Config, log logger.Logger, stats stats.Stats) ([]enricher.PipelineEnricher, error) {
+	var enrichers []enricher.PipelineEnricher
+
+	if conf.GetBool("GeoEnrichment.enabled", false) {
+		log.Infof("Setting up the geolocation pipeline enricher")
+
+		geoEnricher, err := enricher.NewGeoEnricher(conf, log, stats)
+		if err != nil {
+			return nil, fmt.Errorf("starting geo enrichment process for pipeline: %w", err)
+		}
+		enrichers = append(enrichers, geoEnricher)
+	}
+
+	return enrichers, nil
+}
```

Comment:
since this function is being used across app handlers, we could move it out of `embeddedAppHandler.go` to `setup.go`